app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -211,7 +218,7 @@ class SparkRestClient(sparkConf: SparkConf) {
   }
 
   private def getStageDatas(attemptTarget: WebTarget): Seq[StageDataImpl] = {
-    val target = attemptTarget.path("stages")
+    val target = attemptTarget.path("stages/withSummaries")
```

Comment:
This is LinkedIn specific REST endpoint and wont work in open source till it's contributed back to Spark upstream. Probably going ahead we should refactor the code and have our own SparkRestClient implementation. The abstraction for us is primarily at the fetcher level. So probably have a linkedin specific spark fetcher implementation which extends SparkFetcher which currently exists, reuses the part where we are fetching event logs but has custom Spark rest client implementation.

Reply:
Yes, this is LinkedIn specific, and separating out the code would make sense.